pkgs/top-level/all-packages.nix

```diff
@@ -179,7 +179,8 @@ with pkgs;
 
   # `fetchurl' downloads a file from the network.
   fetchurl = import ../build-support/fetchurl {
-    inherit curl stdenv;
+    inherit stdenv;
+    curl = curl.override { gssSupport = false; };
```

Comment:
I would not make a different version for just `fetchurl`. It is pretty likely that we end up with two different libcurl versions anyway since `curl` is one of the most used library.

Reply:
I'm a little worried about putting it in "curl" because it grows the Nix closure. If I make a different version for just "fetchurl", I guess I don't have that problem? I don't want to put it in "curlFull", though, because curlFull isn't used when when derivations want to compile against libcurl.

What I think we should probably be doing is have a "curlBasic" version which is used by fetchurl and other things in the closure of Nix. And then make normal "curl" be "curlFull".

Reply:
It will blow the closure in general. Curl is used anywhere. By having two version of libcurl it will only become worse.

Reply:
The other issue (kind of the same issue) is that if we add Kerberos support to the curl used in fetchurl, those Kerberos dependencies need to use fetchurlboot when building or there will be a circular dependency. Right?

I can do that (change them to use fetchurlboot) if that is desired.

Reply:
I would like to know the opinion of the other maintainers here.